FilePersistence/src/version_control/merge/MergeV2.cpp

```diff
@@ -41,6 +41,11 @@ namespace FilePersistence {
 
 bool MergeV2::commit(const Signature& author, const Signature& committer, const QString& message)
 {
+	if (isAlreadyMerged())
```

Comment:
Please don't check this here. Check it in the tests. Remove all of this new code.